include/clblast_plugin.h

```diff
@@ -101,12 +101,28 @@ class Interface : public Base {
 // (which inherit from Xgemm).
 class Routine : public Base {
  public:
+  Routine();
   virtual ~Routine();
 
   // Custom database entries
   // NOTE: custom database entries completely override built-in ones on a per-routine basis
   // (i. e. if Xgemm/32 is overridden, then the built-in database is never looked up for Xgemm/32)
   std::vector<const database::Routine *> database;
+
+  // Custom OpenCL kernel source, to be used alongside/instead of the built-in OpenCL kernel
+  // NOTE: host<->device API/ABI is not considered stable, so avoid using built-in host routines
+  // with custom kernels.
+  std::string kernel;
+  enum class KernelMode {
+    // Custom kernel is ignored
+    Default = 0,
+
+    // Custom kernel is appended to built-in PadCopyTransform code
+    NeedPCT,
+
+    // Custom kernel replaces all built-in code
+    Custom
+  } kernel_mode;
```

Comment:
Maybe this can be done a bit clearer with less whitespace? I would also add integers to the other enums, at first for me it wasn't clear what these things are, e.g:
```
enum class KernelMode {
   Default = 0, // Custom kernel is ignored
   NeedPCT = 1, // Custom kernel is appended to built-in PadCopyTransform code
   Custom = 2 // Custom kernel replaces all built-in code
 } kernel_mode;
```
Another option is to use the [Hungarian notation](http://stackoverflow.com/questions/1230854/objective-c-why-do-constants-start-with-k), similar to the other enums in CLBlast?